pkcs11/pkcs11.go

```diff
@@ -1711,7 +1707,11 @@ func (s *Slot) generateECDSA(o keyOptions) (crypto.PrivateKey, error) {
 	return priv, nil
 }
 
-func (r *rsaPrivateKey) WithHash(hash crypto.Hash) (*rsaPrivateKey, error) {
+func WithHash(privKey crypto.PrivateKey, hash crypto.Hash) (crypto.PrivateKey, error) {
+	r, ok := privKey.(*rsaPrivateKey)
+	if !ok {
+		return nil, fmt.Errorf("expected type rsaPrivateKey, but got type %T",privKey)
```

Comment:
I'm wondering if you should have a separate PR that runs `go fmt` on all the files touched. There seems to be a missing space before `privKey`.

Reply:
I committed gofmt changes on this PR, if that is okay.

Reply:
In the future it should be an isolated commit. Otherwise it makes it hard to see what changes are formatting vs real change

Reply:
Ahh okay, that makes sense